sig/datadog/core/telemetry/event/synth_app_client_configuration_change.rbs

```diff
@@ -3,9 +3,13 @@ module Datadog
     module Telemetry
       module Event
         class SynthAppClientConfigurationChange < AppStarted
-          def type: () -> "app-client-configuration-change"
+          def type: -> ("app-client-configuration-change" | "app-started")
```

Comment:
TBH I think it should be unspecific instead, because the point is to give type, not the value

```suggestion
          def type: -> ::String
```